desktop/ui/src/main/java/org/datacleaner/windows/AnalysisJobBuilderWindowImpl.java

```diff
@@ -558,6 +582,33 @@ protected boolean onWindowClosing() {
         return windowClosing;
     }
 
+    private boolean isJobUnsaved(FileObject lastSavedJobFile, AnalysisJobBuilder analysisJobBuilder) {
+        if (lastSavedJobFile == null) {
+            return true;
+        }
+
+        InputStream lastSavedOutputStream = null;
+        ByteArrayOutputStream currentOutputStream = null;
+        try {
+            lastSavedOutputStream = getJobFile().getContent().getInputStream();
+            JaxbJobWriter writer = new JaxbJobWriter(_configuration);
+            currentOutputStream = new ByteArrayOutputStream();
+            writer.write(_analysisJobBuilder.toAnalysisJob(false), currentOutputStream);
+            String currentJob = new String(currentOutputStream.toByteArray());
+            String currentJobNoMetadata = currentJob.replaceAll("\n", "").replaceAll("<job-metadata>.*</job-metadata>", "");
+            String lastSavedJob = IOUtils.toString(lastSavedOutputStream);
```

Comment:
We have a similar helper thingie in FileHelper.readFileAsString(...). Not that it matters much, but I guess we already use FileHelper in a lot of places so it might be better?